### PR TITLE
optimize: upgrade logback dependency to 1.2.9

### LINKED ISSUE #4634: logback-classic-1.2.0
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
Logback是一个 Java 应用程序的日志记录框架。
Logback存在远程代码执行漏洞，有编辑配置文件所需权限的攻击者可以制作恶意配置，允许执行从 LDAP 服务器加载的任意代码。

版本比对检测原理：检查当前系统中logback-classic版本是否在受影响版本内|版本比对检测结果：- logback-classic
当前安装版本：1.2.0
应用相关信息：
- 应用路径：/data/soft/seata/seata/seata-server-1.4.2/lib/logback-classic-1.2.0.jar
该主机存在此漏洞

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅲ. Describe what you expected to happen
将 Logback  升级到 1.2.9 及以上版本，下载地址：https://logback.qos.ch/download.html

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :  1.8
- Seata version:  1.4.2
- OS : liunx
- Others:

### PR DESCRIPTION
**Pull request by Smery-lxm**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?

fixes #4634

### CODE REVIEW

File: dependencies/pom.xml [hunk 1]
```diff
@@ -587,6 +588,17 @@
                 <artifactId>simpleclient_httpserver</artifactId>
                 <version>${prometheus.client.version}</version>
             </dependency>
+            <!--logback -->
+            <dependency>
+                <groupId>ch.qos.logback</groupId>
```
Comment (a364176773) [inline dependencies/pom.xml:593]: server模块的pom中是否也应该引入这俩依赖?

Comment (Smery-lxm) [reply dependencies/pom.xml:593]: server模块会识别并替换成1.2.9版本

Comment (Smery-lxm) [reply dependencies/pom.xml:592]: 已添加进server模块

**Review by slievrly — COMMENTED** [review]
why revert previous commit?

**Comment by Smery-lxm** [comment]
> why revert previous commit?

I clicked merge branch then I thought it may be a mistake T.T

**Comment by slievrly** [comment]
> > why revert previous commit?
> 
> I clicked merge branch then I thought it may be a mistake T.T

keep the change only your want to modify in https://github.com/seata/seata/pull/4728/files

**Comment by Smery-lxm** [comment]
> > > why revert previous commit?
> > 
> > 
> > I clicked merge branch then I thought it may be a mistake T.T
> 
> keep the change only your want to modify in https://github.com/seata/seata/pull/4728/files

Ok. I used force-push to modify what I want to do

**Review by a364176773 — APPROVED** [review]
LGTM 请在changes文件夹中的中英两版develop.md中登记pr和作者信息,参考已登记信息填写即可

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**